bytecode/instr.py

```diff
@@ -282,6 +313,27 @@ def stack_effect(self, jump=None):
         else:
             return dis.stack_effect(self._opcode, arg, jump=jump)
 
+    def pre_and_post_stack_effect(self, jump=None):
+
+        _effect = self.stack_effect(jump=jump)
```

Comment:
FOR_ITER needs also a special case since it adds to the stack based on TOS for a jump not taken and pop TOS when jumping

Reply:
actually I don't think this needs special consideration since the stack effect of  `FOR_ITER` is push 1 when no jump and pop 1 when jump, since the stack effect of non-pushing backs are all returned as "pre effect", this should be just fine.

Reply:
But in the first case you need something of the stack so for your analysis to hold you would need a pre effect of -1 and a post effect of 2, no ?

Reply:
ah yea, that makes sense then, thank you for the clarification